addons/pos_paytm/static/src/js/payment_paytm.js

```diff
@@ -0,0 +1,160 @@
+                );
+                if (data?.error) {
+                    throw data?.error;
+                }
+                let resultCode = data?.resultCode;
+                if (resultCode === 'S' && data?.merchantReferenceNo != referenceId){
+                    throw this.env._t("Reference number mismatched");
+                } else if (resultCode === 'S') {
+                    paymentLine.paytm_authcode = data?.authCode;
+                    paymentLine.paytm_issuer_card_no = data?.issuerMaskCardNo;
+                    paymentLine.paytm_issuer_bank = data?.issuingBankName;
+                    paymentLine.paytm_payment_method = data?.payMethod;
+                    paymentLine.card_type = data?.cardType;
+                    paymentLine.paytm_card_scheme = data?.cardScheme;
+                    paymentLine.paytm_reference_no = data?.merchantReferenceNo;
+                    paymentLine.transactionId = data?.merchantTransactionId;
+                    paymentLine.payment_date = data?.transactionDateTime;
+                    return resolve(data);
+                } else {
+                    this.pollTimeout = setTimeout(fetchPaymentStatus, REQUEST_TIMEOUT, resolve, reject);
+                }
+            } catch (error) {
+                const order = this.pos.selectedOrder;
+                this._incrementRetry(order.uid);
+                paymentLine.set_payment_status('force_done');
+                this._showError(error, 'paytmFetchPaymentStatus');
+                return resolve(false)
+            };
+        };
+        return new Promise(fetchPaymentStatus);
+    }
+    /**
+     * @param { float } amount
+     * @param { string } transactionId
+     * @param { string } referenceId
+     * @param { datetime } timestamp
+     * @returns Promise
+    */
+    async makePaymentRequest(amount, transactionId, referenceId, timestamp) {
+        try {
+            const data = await this.env.services.orm.silent.call(
+                'pos.payment.method',
+                'paytm_make_payment_request',
+                [[this.payment_method.id], amount, transactionId, referenceId, timestamp],
+            );
+            if (data?.error) {
+                throw data.error;
+            }
+            return data;
+        } catch (error) {
+            this._showError(error, 'paytmMakePaymentRequest');
+            return false;
+        };
+    }
+
+    // ---------------------------------------------------------------------------
+    // Private methods
+    // ---------------------------------------------------------------------------
+
+    _retryCountUtility(uid, remove=false) {
+        if(remove){
+            localStorage.removeItem(uid);
+        }
+        else{
+            return localStorage.getItem(uid) || (localStorage.setItem(uid, 0) && 0);
+        }
+    }
+
+    _incrementRetry(uid) {
+        let retry = localStorage.getItem(uid);
+        localStorage.setItem(uid, ++retry);
+    }
```

Comment:
Why use the localstorage to register a counter ?
Just use a global variable

Reply:
Doesn't seem much working because when in case of failure if the payment is done by cash/anyother method other than the paytm the retry doesn't resets to 0